libknet/link.c

```diff
@@ -298,6 +343,16 @@ int knet_link_set_enable(knet_handle_t knet_h, uint16_t host_id, uint8_t link_id
 	}
 
 	if (enabled) {
+		if (knet_h->transport_ops[link->transport_type]->link_allocate(
+			    knet_h, knet_h->transports[link->transport_type],
+			    link,
+			    &link->transport, link_id,
+			    &link->src_addr, &link->dst_addr,
+			    &link->outsock) < 0) {
+			err = -1;
```

Comment:
looks like it´s missing savederrno = errno here. I assume errno it´s coming from link_allocate.